### DISCUSSION TITLE
git push/pull like approach for script or other objects

### DISCUSSION THREAD

**Discussion by marcel-more** [fmIDE/fmIDE#19] [Ideas]
Yesterday we already discussed the idea of a simple upgrade-mechanism with the help of 'insert from URL' inside fmIDE.
It would be great to develop some kind of mechanism, to store and retrieve 'objects' from github.

a) This could be simple 'Snippets' of code - e.g. a Script or Custom Function that is inserted into an own solution. Think of it like a kind of Code-Sharing-Plattform for 'addon' like features and functions.

b) To think this further, the mechanism could allow to update stored objects on github from the original source inside a FileMaker solution. With a push of a button / starting a script / hotkey whatever... the object is copied from FileMaker as XML representation and committed to a git repo. Together with a commit-message of course. Then later it could be distributed / updated from the repo to other deployments of the same solution.

First goal could be an auto-updater for the fmIDE script itself. So you can keep the tool inside your solutions uptodate automagically.

**Comment by mrwatson-de** [comment]
fmIDE = ✨ automagically ✨ 😁

**Comment by mrwatson-de** [comment]
On the topic of GitHub / Gist-based object / snippet store, I've also been thinking about the question ... How can we use GitHub's Pull-Request and merging functionality better, given that FileMaker files are a binary blob, and do *not* play ball.

One idea, was that we could not only have the FileMaker file in the GitHub store, but also a text rendition of the file content - in this case particularly the fmIDE script - which is always exported before committing so that it remains synchronous with the script in the blob file. 

Like that when we commit a new version of this metadata...

- GitHub will merge the text-rendition of the script
- alert us of any conflicts,
- give us the possibility to compare before and after to see exactly what has changed,
- which in turn will aid us in actually patching the master FileMaker file from the PR (Pull Request) File

^^^ It's the start of an idea ^^^ 😆
